### ISSUE TITLE
Disappearance of checkboxes with release v4.1.9

### ISSUE THREAD

**Issue by Sandrine-D** [bolt/core#2263]
Since the upgrade in v4.1.9 all my checkboxs have disappeared in contenttype editor. This happens for all contenttypes I have. 

A sample of my contenttype:
```
top:
    label: "Afficher sur la page d'accueil?"
    type: checkbox
    default: checked
    group: Options
menu:
    label: "Afficher dans le top menu?"
    type: checkbox
    default: checked
titre_menu:
    label: "Titre top menu"
    type: text
    localize: true
```

Both checkboxes fields "top" and "menu" are missing. There is no error/exception.

<img width="753" alt="Capture d’écran 2020-12-22 à 12 23 32" src="https://user-images.githubusercontent.com/74857683/102932585-fe3d4980-44b9-11eb-9ecc-f4096c4dc508.png">

This worked previously with 4.1.5.

**Comment by bobdenotter** [comment]
That’s weird. Does the browsers console / inspector say anything or give errors?

can you run composer update again? Does it run fully without errors?

**Comment by Sandrine-D** [comment]
The first time I had some issues with `composer update`.
```
Fatal error: Uncaught Error: Maximum function nesting level of '256' reached, aborting! in phar:///opt/cpanel/composer/bin/composer/src/Composer/Config.php on line 220

Error: Maximum function nesting level of '256' reached, aborting! in phar:///opt/cpanel/composer/bin/composer/src/Composer/Config.php on line 220
```

So I tried `php composer.phar update` which worked fine.
Then I relaunched `composer update` which was now ok: Nothing to install or update.

But I think I found something:

In the **Chrome** browser inspector, **which failed**, I have this:
`<div id="field-checkbox-menu" class="form-group    is-normal"><editor-checkbox id="fields[menu]" value="1" name="fields[menu]" label="Afficher dans le top menu?" mode="checkbox"></editor-checkbox></div>`

I have some scripts warnings:
`0.js:50 [Violation] Added synchronous DOM mutation listener to a 'DOMNodeInserted' event. Consider using MutationObserver to make the page more responsive.`
`redactor.min.js:1  [Violation] Added non-passive event listener to a scroll-blocking 'touchstart' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952`

And **it works fine with Safari:**
`<div id="field-checkbox-menu" class="form-group    is-normal"><div><div class="custom-control custom-checkbox"><input id="fields[menu]" name="fields[menu]" type="checkbox" class="custom-control-input"> <label for="fields[menu]" class="custom-control-label">Afficher dans le top menu?</label> <input type="hidden" name="fields[menu]" value="1"></div></div></div>`

So it seems to be a JS script issue with Chrome.
I don't have other browsers available to test know.

**Comment by bobdenotter** [comment]
I've not been able to reproduce this.. Could you verify it with a simple contenttype like this: 

```
dummies:
    name: Dummies
    singular_name: Dummy
    fields:
        title:
            type: text
        box:
            type: checkbox
```

In Firefox, Chrome and Safari it works as expected: 

![Screenshot 2020-12-23 at 17 11 59](https://user-images.githubusercontent.com/1833361/103016227-0b058000-4542-11eb-8a21-a486e13d6d89.png)


Alternatively, since you're upgrading: make sure you have the latest assets installed. When running `composer update`, do you get _no_ errors? And you see these lines? 

```
Executing script assets:install --symlink --relative public [OK]
Executing script bolt:copy-assets [OK]
```

Also, you can verify if the versions match by running: `composer show | grep bolt`

![Screenshot 2020-12-23 at 17 15 17](https://user-images.githubusercontent.com/1833361/103016517-74858e80-4542-11eb-85a8-6280fc9a5c52.png)


Both `bolt/core` and `bolt/assets` should say `4.1.9`.

**Comment by Sandrine-D** [comment]
It works, but I don't really know why!

I just added your simple content type test without modifying anything else and everything is back to normal. All my checkboxes reappeared.

The warnings are still present in the browser's inspector, so it's not related to that. Maybe I just needed to save my contenttype.yml file again after the update? ... Strange!

**Comment by Sandrine-D** [comment]
I'm back again!! 
After a quick check, all my checkboxes are now selected! 

Have you change something with checkboxes between v4.1.5 and v4.1.9? It is seems checkboxes value is now managed in a different way.

**Comment by Sandrine-D** [comment]
I investigated and I found the issue.
Previously, before bolt update, checkboxes values were registered with "0" and "1" in the database. During the bug, after upgraded bolt in v4.1.9 and before saving again the contenttype file, values were registered with 0 and 1 (without quote). Now it is "true" or "false".

So I updated all impacted fields in the database and the twig queries and everything is now OK.
